src/bindgen/cargo/cargo.rs

```diff
@@ -88,86 +89,68 @@ impl Cargo {
-        // look for that too.
-        let mut dependency_name = dependency_name;
-        let mut replaced_name = dependency_name.replace("_", "-");
-
-        // The Cargo.toml may contain a rename which we need to apply
-        for metadata_package in &self.metadata.packages {
-            if metadata_package.name == package.name {
-                for dep in &metadata_package.dependencies {
-                    if let Some(ref rename) = dep.rename {
-                        if rename == dependency_name || rename == &replaced_name {
-                            dependency_name = &dep.name;
-                            replaced_name = dependency_name.replace("_", "-");
-                        }
-                    }
-                }
-            }
-        }
+        let mut dependencies = None;
 
+        // Find the dependencies listing in the lockfile
         if let &Some(ref root) = &lock.root {
             if root.name == package.name && root.version == package.version {
-                if let Some(ref deps) = root.dependencies {
-                    for dep in deps {
-                        let (name, version) = parse_dep_string(dep);
-
-                        if name == dependency_name || name == &replaced_name {
-                            return Some(PackageRef {
-                                name: name.to_owned(),
-                                version: version.to_owned(),
-                            });
-                        }
-                    }
-                }
-                return None;
+                dependencies = root.dependencies.as_ref();
             }
         }
-
         if let &Some(ref lock_packages) = &lock.package {
             for lock_package in lock_packages {
                 if lock_package.name == package.name && lock_package.version == package.version {
-                    if let Some(ref deps) = lock_package.dependencies {
-                        for dep in deps {
-                            let (name, version) = parse_dep_string(dep);
-
-                            if name == dependency_name || name == &replaced_name {
-                                return Some(PackageRef {
-                                    name: name.to_owned(),
-                                    version: version.to_owned(),
-                                });
+                    dependencies = lock_package.dependencies.as_ref();
+                    break;
+                }
+            }
+        }
+
+        let mut output = vec![];
+
+        if let Some(dependencies) = dependencies {
+            for dep in dependencies {
+                let (dep_name, dep_version) = parse_dep_string(dep);
+
+                // Try to find the cfgs in the Cargo.toml
+                let mut cfg = None;
+                'outer: for meta_package in &self.metadata.packages {
+                    if meta_package.name == package.name && meta_package.version == package.version
```

Comment:
Could use `find` as well and remove the loop labeling.

Reply:
Also, this is O(n^2) or worse right? That's not great... The old code was only ran when `should_parse_dependency` returned true, so it almost never ran. Though I guess if perf in Gecko is still acceptable it's probably not a big deal to optimize for this? I guess you'll get some build-time regressions if this really matters.